### ISSUE TITLE
Failed to convert to PMML with error ContinuousLabel cannot be cast to CategoricalLabel

### ISSUE THREAD

**Issue by ly54mm** [jpmml/sklearn2pmml#157]
[github.zip](https://github.com/jpmml/sklearn2pmml/files/3213326/github.zip)
Hi,
I want to convert my xgboost model to pmml, but I get the following error when calling
```
sklearn2pmml(pipeline, "XGBoost.pmml")
java.lang.ClassCastException: org.jpmml.converter.ContinuousLabel cannot be cast to org.jpmml.converter.CategoricalLabel
```
I have my code and test input file attached. Any ideas of how to fix it? 
[github_code.zip](https://github.com/jpmml/sklearn2pmml/files/3213442/github_code.zip)

**Comment by vruusmann** [comment]
You're using a objective function (on line 16), which is incompatible with `XGBRegressor` (on line 21).

If you change `param = {'objective': 'binary:logistic'}` (logistic classification, categorical label) to `param = {'objective': 'reg:logistic'}` (logistic regression, continuous label), then everything works as advertised.

Why don't you switch to "normal" pipeline-oriented workflow? It would be much conciser than dealing with train and test DMatrix objects manually.
